main.cpp

```diff
@@ -27,41 +28,80 @@ using namespace std;
 
 void brute_angles(Platform* plat, const Vec3f& m_pos, const float spd, const Vec3f & normals, const Mat4& trans) {
 	//iterate over hau instead of sticks
-	for (int hau = 0; hau < 65535; hau += 16) {
-		if (abs((short)(int)(m_pos[0] + gSineTable[hau >> 4] * normals[1] * (spd / 4.0f))) >= 8192) {
+#ifdef __AVX2__
+	const __m256  m_pos_vector = _mm256_set1_ps(m_pos[0]);
+	const __m256 normal_vector = _mm256_set1_ps(normals[1]);
+	const __m256 speed_vector = _mm256_set1_ps(spd / 4.0f);
+	const __m256 speed_x_normal = _mm256_mul_ps(normal_vector, speed_vector);
+#endif
+	for (int hau1 = 0; hau1 < 65535; hau1 += 128) {
+#ifdef __AVX2__
+		__m256 sin_vector = _mm256_loadu_ps(gSineTable.data() + (hau1 >> 4));
+		__m256 float_check = _mm256_fmadd_ps(speed_x_normal, sin_vector, m_pos_vector);

```

Comment:
it may be okay in practice, but technically this changes rounding. same thing with pre-multiplying the normal with spd/4